src/Extension/FluentExtension.php

```diff
@@ -679,6 +683,91 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
+
+        // We don't have any localised relations to cover
+        if (count($copyHasOneRelations) === 0) {
+            return;
+        }
+
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        // We don't have a current locale which indicates that the Fluent setup is incomplete - bail out
+        if (!$currentLocale) {
+            return;
+        }
+
+        $locales = $this->getLocaleCodesForModel($owner);
+
+        // Current locale can be skipped as it was already handled correctly
+        $locales = array_diff($locales, [$currentLocale]);
+
+        // No locales need to be actioned
+        if (count($locales) === 0) {
+            return;
+        }
+
+        $ownerVersioned = $owner->hasExtension(Versioned::class);
+
+        foreach ($locales as $locale) {
+            FluentState::singleton()->withState(
+                static function (FluentState $state) use ($owner, $original, $ownerVersioned, $locale, $copyRelations): void {
+                    $state->setLocale($locale);
+
+                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);
+                    $localisedOriginal = DataObject::get($original->ClassName)->byID($original->ID);
+
+                    // Couldn't find localised data to work with
+                    if (!$localisedOwner || !$localisedOriginal) {
+                        return;
+                    }
+
+                    // Duplicate all localised relations
+                    foreach ($copyRelations as $relation) {
+                        $localisedRelation = $localisedOwner->getComponent($relation);
+                        $originalRelation = $localisedOriginal->getComponent($relation);
+
+                        if (!$originalRelation instanceof DataObject) {
+                            continue;
+                        }
+
+                        if (!$originalRelation->isInDB()) {
+                            continue;
+                        }
+
+                        // Determine if Top page ID information is available
+                        $topPageID = (int) $localisedRelation->hasField('TopPageID')
+                            ? $localisedRelation->TopPageID
+                            : 0;
+
+                        // Elemental module compatibility
+                        if ($topPageID && $originalRelation->hasMethod('withFixedTopPage')) {
+                            $duplicate = $originalRelation->withFixedTopPage(
+                                $topPageID,
+                                static function () use ($originalRelation): DataObject {
+                                    // Duplicate needs to include write to store the top page data
+                                    return $originalRelation->duplicate();
+                                }
+                            );
+                        } else {
+                            $duplicate = $originalRelation->duplicate();
```

Comment:
```suggestion
                            $duplicate = $originalRelation->duplicate(false);
```
There's no comment on this one so I assume we don't need an explicit write here and can rely on `setComponent` calling `write()` as per https://github.com/tractorcow-farm/silverstripe-fluent/pull/941#discussion_r2013304006